documentatie/flowcharts/date-entry-navigation.md

```diff
@@ -0,0 +1,149 @@
+
+    show-error
+    error-icon-next-page
+    %% error-icon-prev-page is illegal state
+    error-icon-cur-page
+
+    show-warning
+    warning-icon-next-page
+    warning-icon-prev-page
+    warning-icon-cur-page
+
+    error-next-page{error-next-page?}
+    error-prev-page{error-prev-page?}
+    error-cur-page{error-cur-page?}
+
+    warning-next-page{warning-next-page?}
+    warning-prev-page{warning-prev-page?}
+    warning-cur-page{warning-cur-page?}
+
+    cur-page-submitted{cur-page-submitted?}
+
+    fill-cached-input
+
+    %% flow
+    flow-start --> error-prev-page
+
+    error-prev-page -- yes --> illegal-state
+    error-prev-page -- no --> error-next-page
+
+    error-next-page -- yes --> error-icon-next-page
+    error-icon-next-page --> error-cur-page
+    error-next-page -- no --> error-cur-page
+
+    error-cur-page -- yes --> cur-page-submitted
+    cur-page-submitted -- yes --> error-icon-cur-page
+    error-icon-cur-page --> show-error
+    cur-page-submitted -- no --> warning-prev-page
+    error-cur-page -- no --> warning-prev-page
+
+    warning-prev-page -- no --> warning-next-page
+    warning-prev-page -- yes --> warning-icon-prev-page
+    warning-icon-prev-page --> warning-next-page
+
+    warning-next-page -- no --> warning-cur-page
+    warning-next-page -- yes --> warning-icon-next-page
+    warning-icon-next-page --> warning-cur-page
+
+    warning-cur-page -- no --> fill-cached-input
+    warning-cur-page -- yes --> warning-icon-cur-page
+    warning-icon-cur-page --> show-warning
+
+    show-error --> fill-cached-input
+    show-warning --> fill-cached-input
+
+    fill-cached-input --> flow-end
+
+```
+
+
+## Click "Volgende" and call API
+
+```mermaid
+flowchart TD
+    %% elements
+    flow-start([start])
+    flow-end([end])
+
+    click-next
+    call-api
+    go-to-prev-page
+    go-to-next-page
+    render-cur-page
+
+    error-prev-page{error-prev-page?}
+    error-cur-page{error-cur-page?}
+    error-next-page{error-next-page?}
+    warnings{warnings?}
+
+    %% flow
+
+    flow-start --> click-next
+    click-next --> call-api
+    call-api --> error-prev-page
+    error-prev-page -- yes --> go-to-prev-page
+    go-to-prev-page --> flow-end
+
+    error-prev-page -- no --> error-cur-page
+
+    error-cur-page -- yes --> render-cur-page
+    error-cur-page -- no --> error-next-page
+
+    error-next-page -- yes --> warnings
+    error-next-page -- no --> warnings
+
+    warnings -- no --> go-to-next-page
+    warnings -- yes --> render-cur-page
+    
+    go-to-next-page --> flow-end
+    render-cur-page --> flow-end
+
+```
+
+
+## Click nav item
+
+```mermaid
+flowchart TD
+    %% elements
+    flow-start([start])
+    flow-end([end])
+
+    click-nav-item
+    save-changes{save-changes?}
+    go-to-page
+    remain-on-page
+    api-call
+
+    errors-or-warnings-cur-page{errors-or-warnings-cur-page?}
+
+    %% flow
+    flow-start --> click-nav-item
+    click-nav-item --> save-changes
```

Comment:
not sure anymore whether this was what we agreed upon. This is rather nasty: If I want to navigate to a previous page, because I just discovered I did something wrong there, this logic will force me to complete the current page before I can do that. Not sure we would want this.

Reply:
It's how I interpreted https://github.com/kiesraad/abacus/issues/50#issuecomment-2210730723 `save-changes` is the dialog asking the user if they want to save their changes or not. So the user does have the choice to not save their changes and go to the other page.

What is missing in this flowchart is that if the user does not wish to save changes, we should cache the input.

Which means we need to be mindful of our choice of wording in the [save changes dialog](https://www.figma.com/design/zZlFr8tYiRyp4I26sh6eqp/Kiesraad---Abacus-optelsoftware?node-id=3151-21014&t=Qt2ouMkmbcb1VKOR-0).

Reply:
@jschuurk-kr ah, that's not how https://github.com/kiesraad/abacus/issues/50#issuecomment-2210730723 was meant. There is a subtle difference between the following cases:

- you are on the furtest page and navigate to any previous page: we save your input, but don't do any validation. This input has an 'incomplete' state, which is fine for the furtest page. Valdiation only happens when you use the 'volgende' button and try to continue to the next step. 
- you are not on the furtest page, have made changes and navigate to any other page: because your input might contain errors that will make any next step an illegal one, we need to validate the input. Now we show the dialog to ask the user whether they want to save their changes.

Shorter:
- When user uses navigation:
  - on furtest page? Submit/cache input, no validation
  - not on furtest page? Aks user if they want to save changes
    - y: submit to backend, validate. Errors or warnings?
      - remain on page, show error/warning and have user fix them or accept warnings
    - n: reset changes, navigate to requested page.

Reply:
flowchart aangepast